test/button.test.tsx

```diff
@@ -1,11 +1,265 @@
+      await render(<MuiButton endIcon={<span>icon</span>}>EndIcon Button</MuiButton>)
+    })
+    .assertion(button.has({ className: including("MuiButton-root") }))
+    .assertion(button.has({ className: including("MuiButton-text") }))
+    .assertion(button.has({ iconLabel: including("MuiButton-endIcon") }))
+    .assertion(button.has({ iconLabel: not(including("MuiButton-startIcon")) }))
+  )
+  .child('button renders with ripple by default', (test) => test
+    .step('render button with ripple', async() => {
+      await render(<MuiButton TouchRippleProps={{ className: 'touch-ripple'}}>Ripple Button</MuiButton>)
+    })
+    .step('has root span with custom css ripple', async() => {
+      await button.has({ customRippleClass: including('touch-ripple')})
+    })
+  )
+  .child('can disable ripple effect', (test) => test
+    .step('render button disabled ripple', async() => {
+      await render(<MuiButton disableRipple TouchRippleProps={{ className: 'touch-ripple'}}>Ripple Button</MuiButton>)
+    })
+    .step('span with custom ripple class', async() => {
+      await button.has({ rippleEnabled: false})
+    })
+  )
+  .child('can disable elevation', (test) => test
+    .step('render button with disabled elevation', async() => {
+      await render(<MuiButton disableElevation>Disabled Elevation Button</MuiButton>)
+    })
+    .assertion(button.has({ className: including("MuiButton-disableElevation") }))
+  )
+  .child('have a focusRipple by default', (test) => test
+    .step('render button', async() => {
+      await render(<MuiButton TouchRippleProps={{ classes: { ripplePulsate: 'pulstat-focus-visible'} }}>Focus ripple default button</MuiButton>)
+    })
+    .step('focus button', async() => {
+      await button.focus()
+    })
+    .step('has span with css ripple', async() => {
+      await button.has({ customRippleClass: including('pulstat-focus-visible')})
+    })
+  )
+  .child('can disable focusRipple', (test) => test
+    .step('render button', async() => {
+      await render(
+        <MuiButton
+          disableFocusRipple
+          TouchRippleProps={{ classes: { ripplePulsate: 'pulstat-focus-visible'} }}
+        >
+          Focus ripple disabled button
+        </MuiButton>
+      )
+    })
+    .step('focus button', async() => {
+      await button.focus()
+    })
+    .step('should not have visible css ripple', async() => {
+      await button.has({ rippleVisible: false })
+    })
+  )
+  .child('should automatically change button to an anchor when href is provided', (test) => test
+    .step('render as Link', async() => {
+      await render(<MuiButton href="https://google.com">Href Button</MuiButton>)
+    })
+    /**
+     * When a href is passed to the Material Button it will render <a> tag so
+     * we will switch to use a built-in Link Interactor.
+     */
+    .assertion(Link().has(
+     {
+        className: including("MuiButton-root"),
+        href: including("https://google.com")
+      }
+    ))
+  )
```

Comment:
Should we use the `MuiButton` interactor for this form as well? It feels like that'd be better. So the `MuiButton` selector should be expanded match `href` links as well.

Reply:
@jnicklas The MuiButton selector now is based of the base Button interactor from bigtest. Are you saying to do something like this

`.selector('button,input[type=button],input[type=submit],input[type=reset],input[type=image], a')`

right inside the current Material Interactor that we are building?

Reply:
One way would be to to just delegate actions:

```ts
import { HTML } from '@bigtest/interactor';

export default HTML.extend<HTMLButtonElement | HTMLAnchorElement>("MUI Button")
  .selector('.MUIButton-root')
```

You will need to add a `disabled` filter yourself. If there are cases where you have to choose different paths between what type of element you actually have:

```ts
function isButtonElement(element: HTMLButtonElement | HTMLLinkElement) element is HTMLButtonElement {
  return element.tagName.toUpperCase() === 'BUTTON';
}
```

and then you can use it:

```ts
if (isButtonElement) {
  // do this:
else {
  // do that
}
```